# Assembly analysis tools and papers

![GitHub Repo stars](https://img.shields.io/github/stars/Jwindler/Assembly_tools) ![GitHub pull requests](https://img.shields.io/github/issues-pr/Jwindler/Assembly_tools) ![GitHub License](https://img.shields.io/github/license/Jwindler/Assembly_tools)

Genome Assembly tools are added by pipeline. **Welcome contribute and get in touch!**

[Google Group](https://groups.google.com/g/assembly-tools) | [Volunteers](https://github.com/Jwindler/Assembly_tools/blob/main/Volunteers.md)  | [CONTRIBUTING](https://github.com/Jwindler/Assembly_tools/blob/main/CONTRIBUTING.md)



## Table of content

- [Assembly analysis tools and papers](#assembly-analysis-tools-and-papers)
  - [Table of content](#table-of-content)
  - [Survery](#survery)
  - [Conitg](#conitg)
  - [Scaffold](#scaffold)
  - [Review](#review)
  - [Polish](#polish)
  - [Evaluation](#evaluation)





## Survery



| Name        | Introduction                                       | Paper                                                             | Url                                                | Note |
| ----------- | -------------------------------------------------- | ----------------------------------------------------------------- | -------------------------------------------------- | ---- |
| GenomeScope | Fast genome analysis from unassembled short reads. | [*Bioinformatics*](https://doi.org/10.1093/bioinformatics/btx153) | [Github](https://github.com/schatzlab/genomescope) |      |
|             |                                                    |                                                                   |                                                    |      |





## Conitg



| Name    | Introduction                                                                                     | Paper                                                       | Url                                            | Note |
| ------- | ------------------------------------------------------------------------------------------------ | ----------------------------------------------------------- | ---------------------------------------------- | ---- |
| Hifiasm | Hifiasm is a fast haplotype-resolved de novo assembler initially designed for PacBio HiFi reads. | [*Nat Methods*](https://doi.org/10.1038/s41592-020-01056-5) | [Github](https://github.com/chhylp123/hifiasm) |      |
|         |                                                                                                  |                                                             |                                                |      |





## Scaffold



| Name       | Introduction                                                 | Paper                                                        | Url                                               | Note                                            |
| ---------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------- | ----------------------------------------------- |
| 3D-DNA     | Scaffold genome with Hi-C data                               | [*Science*](https://www.science.org/doi/10.1126/science.aal3327) | [Github](https://github.com/aidenlab/3d-dna)      | Use Hi-C data                                   |
| LACHESIS   | Use Hi-C data for ultra-long-range scaffolding of *de novo* genome assemblies. | [*Nature Biotechnology*](https://www.nature.com/articles/nbt.2727) | [Github](https://github.com/shendurelab/LACHESIS) | LACHESIS is no longer being actively developed. |
| SALSA2     | A tool to scaffold long read assemblies with Hi-C data       | [*bioRxiv*](https://www.biorxiv.org/content/10.1101/261149v1) | [Github](https://github.com/marbl/SALSA)          |                                                 |
| YaHS       | Yet another Hi-C scaffolding tool                            | [*Bioinformatics*](https://doi.org/10.1093/bioinformatics/btac808) | [Github](https://github.com/c-zhou/yahs)          | recommend                                       |
| instaGRAAL | Large genome reassembly based on Hi-C data, continuation of GRAAL | [*Nature Communications*](https://www.nature.com/articles/ncomms6695) | [Github](https://github.com/koszullab/instaGRAAL) | NVIDIA graphics card is required                |
| EndHiC     | a fast and easy-to-use Hi-C scaffolding tool                 | [*Quantitative Biology*](https://doi.org/10.48550/arXiv.2111.15411) | [Github](https://github.com/fanagislab/EndHiC)    |                                                 |
| Pin_hic    | A Hi-C scaffolding method                                    | [*BMC Bioinformatics*](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04453-5) | [Github](https://github.com/dfguan/pin_hic)       |                                                 |
| AutoHiC    | A novel genome assembly pipeline based on deep learning      | [*bioRxiv*](https://doi.org/10.1101/2023.08.27.555031)       | [Github](https://github.com/Jwindler/AutoHiC)     | recommend (Deep Learning)                       |
| ALLHiC     | phasing and scaffolding polyploid genomes based on Hi-C data | [*Nature Plants*](https://www.nature.com/articles/s41477-019-0487-8) | [Github](https://github.com/tangerzhang/ALLHiC)   | recommend (Plant)                               |
|            |                                                              |                                                              |                                                   |                                                 |



## Review



| Name     | Introduction                                                                                             | Paper                                                         | Url                                            | Note |
| -------- | -------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------- | ---------------------------------------------- | ---- |
| Juicebox | a point-and-click interface for using Hi-C heatmaps to identify and correct errors in a genome assembly. | [*bioRxiv*](https://www.biorxiv.org/content/10.1101/254797v1) | [Github](https://github.com/aidenlab/Juicebox) |      |
|          |                                                                                                          |                                                               |                                                |      |



## Polish



| Name | Introduction | Paper | Url | Note |
| ---- | ------------ | ----- | --- | ---- |
|      |              |       |     |      |
|      |              |       |     |      |



## Evaluation

**Genome assembly evaluation tools.**

| Name           | Introduction                                                                                                                | Paper                                                                         | Url                                                               | Note              |
| -------------- | --------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------- | ----------------------------------------------------------------- | ----------------- |
| QUAST          | a quality assessment tool for evaluating and comparing genome assemblies.                                                   | [*Bioinformatics*](https://doi.org/10.1093/bioinformatics/btt086)             | [Github](https://github.com/ablab/quast)                          |                   |
| BioNanoAnalyst | a visualisation tool to assess genome assembly quality using BioNano data.                                                  | [*BMC Bioinformatics*](https://bmcbioinformatics.biomedcentral.com/)          | [Github](https://github.com/AppliedBioinformatics/BioNanoAnalyst) | Use BioNano  data |
| CRAQ           | Identification of errors in draft genome assemblies with single-base pair resolution for quality assessment and improvement | [*Nature Communications*](https://www.nature.com/articles/s41467-023-42336-w) | [Github](https://github.com/JiaoLaboratory/CRAQ)                  | Single base scale |









